src/unpoly/classes/request/xhr_renderer.js

```diff
@@ -38,7 +38,8 @@ up.Request.XHRRenderer = class XHRRenderer {
     }
 
     Object.assign(xhr, handlers)
-    xhr.send(this._getPayload())
+
+    Promise.resolve(this._getPayload()).then(xhr.send)
```

Comment:
This introduces a race condition. When this `up.Request` is aborted while waiting for payload processing, we must not send a request. 

To address this we would need to change `up.Request.prototype.load()`, which currently looks like this:

```js
new up.Request.XHRRenderer(this).buildAndSend({ ... })
```

It probably needs to look something this:


```js
let renderer = new up.Request.XHRRenderer(this)

// Wait for async payload processing
await renderer.build({ ... })

// Don't send the request if someone has aborted us while waiting for the renderer
if (this.state === 'loading') renderer.send()
```

We should have a test for this.